azext_edge/edge/providers/orchestration/upgrade.py

```diff
@@ -204,23 +210,26 @@ def _get_resource_map(self) -> IoTOperationsResourceMap:
                 resource_name=self.instance_name,
                 api_version=INSTANCE_7_API
             )
+            # don't deal with bug bash m2's - only released version
+            if self.instance["properties"]["version"] != "0.7.31":
+                raise ArgumentUsageError(error_msg)
             return self.instances.get_resource_map(self.instance)
         except HttpResponseError:
-            self.require_instance_upgrade = False
+            pass
```

Comment:
Is there any kind of additional validation we should do here to make sure it's the right HttpResponseError?

I also know that `try...except...pass` is gonna flag our static analysis as a general no-no but given how this works I'm not sure what else we should do here

Reply:
well now not really - I can log things if you want